Modix/Modules/InfractionModule.cs

```diff
@@ -65,15 +64,15 @@ public InfractionModule(IModerationService moderationService)
             var replyBuilder = new StringBuilder();
             foreach (var line in tableText.Split("\r\n"))
             {
-                if((replyBuilder.Length + line.Length) > 1998)
+                if ((replyBuilder.Length + line.Length) > 1990)
```

Comment:
In order to avoid the current error, you need to guarantee that the length of the string builder plus the length of the code format characters do not exceed 2000

This probably means truncating the message for each infraction to a restricted length so that the sum of all fields on a line won't exceed a known amount, then restricting the table to a limited number of entries.